crates/core/src/table/parquet_config.rs

```diff
@@ -0,0 +1,166 @@
+// Utilities to represent TableParquetOptions as a set of key-value pairs
+// Only active when the "datafusion" feature is enabled
+
+use std::collections::HashMap;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use std::fmt;
+use std::fmt::Display;
+use datafusion::common::{DataFusionError, file_options::parquet_writer::ParquetWriterOptions};
+use datafusion::config::{ConfigEntry, ConfigField, TableParquetOptions, Visit};
+use parquet::file::properties::WriterProperties;
+
+// String representation of the Parquet configuration.
+// This hashmap holds key-value pairs for the settings in
+// datafusion_common::config::TableParquetOptions
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ParquetConfig {
+    pub table_parquet_options: HashMap<String, String>,
+}
+
+impl Default for ParquetConfig {
+    fn default() -> Self {
+        Self {
+            table_parquet_options: HashMap::new(),
+        }
+    }
+}
+
+impl FromStr for ParquetConfig {
+    type Err = String; // Should hold a project standard error type?
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let deserialized: ParquetConfig = serde_json::from_str(&s)
+            .map_err(|e| format!("Failed to deserialize ParquetConfig: {}", e))?;;
+        Ok(deserialized)
+    }
+}
+
+// Implementing display also implements to_string
+impl fmt::Display for ParquetConfig {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let serialized = serde_json::to_string(&self).unwrap();
+        write!(f, "{}", serialized)
+    }
+}
+
+
+// Function to convert TableParquetOptions to a vector of ConfigEntry
+// This should probably be part of DataFusion's config module but is placed here for now.
+pub fn entries(table_parquet_options: &TableParquetOptions) -> Vec<ConfigEntry> {
+    struct Visitor(Vec<ConfigEntry>);
+
+    impl Visit for Visitor {
+        fn some<V: Display>(
+            &mut self,
+            key: &str,
+            value: V,
+            description: &'static str,
+        ) {
+            self.0.push(ConfigEntry {
+                key: key.to_string(),
+                value: Some(value.to_string()),
+                description,
+            })
+        }
+
+        fn none(&mut self, key: &str, description: &'static str) {
+            self.0.push(ConfigEntry {
+                key: key.to_string(),
+                value: None,
+                description,
+            })
+        }
+    }
+
+    let mut v = Visitor(vec![]);
+    table_parquet_options.visit(&mut v, "", "");
+    
+    // Work around datafusion bug in file_encryption.visit.
+    // Visitor skips column_encryption_properties
+    let mut c = Visitor(vec![]);
+    if let Some(file_encryption) = &table_parquet_options.crypto.file_encryption {
+        file_encryption.column_encryption_properties.visit(&mut c, ".crypto.file_encryption", "");
+        v.0.append(&mut c.0);
+    };
+    
+    v.0
+}
```

Comment:
I should probably add this to datafusion when I go to fix the buggy visitor operation.